pkg/ddl/index.go

```diff
@@ -2415,6 +2447,17 @@ func writeChunk(
 	if restore {
 		restoreDataBuf = make([]types.Datum, len(c.HandleOutputOffsets))
 	}
+
+	indexConditionChecker := make([]func(row chunk.Row) (bool, error), len(indexes))
+	for i, index := range indexes {
```

Comment:
If there is only one index and the condition is pushed down to TiKV, we don't need to check the condition again here.